### ISSUE TITLE
The three readers of a neighbour about to write look under one namespace only, so two copies of one rule take the run in turns

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#710]
### What minimal example or steps are needed to reproduce the bug?

Every namespace reads a plain CSS file, so one rule can be listed twice over the same stylesheet — once under the core's name and once under a syntax's. The three readers that ask what a neighbour is about to write — `writesSharedRun`, `whitespaceAsked` and `writesRunBehindBrace` — resolve that neighbour through `addNamespace(name, syntax.namespace)`, so each of them only ever sees the copy registered under the asking rule's own namespace. The copy under the other namespace writes the same run and is invisible.

```css
/* input */
a {} b {}
```

```shell
# `block-closing-brace-space-after: "never"` first, then `scss/block-closing-brace-newline-after: "always"`
--fix leaves `a {}\nb {}` and reports nothing; a check of that output reports the space rule

# the same two the other way round
--fix leaves `a {}b {}` and reports nothing; a check reports the scss break rule
```

Both rules write, the later-listed one wins, and the loser's warning is dropped as fixed. That is [#704](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/704) exactly, in the one place built to prevent it.

Measured on `945564a`, before `block-closing-brace-space-after` had a fixer: both orders leave `a {}\n b {}` and report the space rule, since only one of the two wrote.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/block-closing-brace-space-after": "never",
    "@stylistic/scss/block-closing-brace-newline-after": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

A reader that asks what a neighbour is about to write finds that neighbour wherever it is listed, since a plain CSS file is read by every namespace.

### What actually happened?

It finds only the copy under its own namespace, so the two copies take the run in turns and whichever is listed last wins in silence.

### Do you have a proposal to fix the bug?

The reading lives in one place, `neighbourSettings`: it matches a configured name against `addNamespace(rule.name, syntax.namespace)` and nothing else. Widening it to every namespace that reads the root would answer for all three of its callers at once, and the run order it returns already ranks a namespace behind the core's through `linenessRank`.

What has to be measured before that is what it costs the callers that read a neighbour's *option* rather than its write — `whitespaceAsked` picks the later-listed live rule, and a second namespace's copy would then be able to win.

Found while writing the fixer of #698, which is what makes the pair above reach the hole; `writesSharedRun` and `whitespaceAsked` have carried it since they were written.
